aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -929,57 +929,61 @@ internal class RealAWSCognitoAuthPlugin(
     ) {
         val forceRefresh = options.forceRefresh
         authStateMachine.getCurrentState { authState ->
-            when (val authZState = authState.authZState) {
-                is AuthorizationState.Configured -> {
-                    authStateMachine.send(AuthorizationEvent(AuthorizationEvent.EventType.FetchUnAuthSession))
```

Comment:
We need this. If signedOut, we need to fetch guest credentials. I think the problem might be that we should be returning guest credentials instead of invalid state exception.